test/acceptance/prioritized-widgets.spec.js

```diff
@@ -0,0 +1,202 @@
+// Prioritized Widget Tests
+describe('Prioritized widgets', function () {
+  beforeEach(function () {
+    localStorage.clear();
+    sessionStorage.clear();
+    pathfora.clearAll();
+    window.lio = {};
+  });
+
+  afterEach(function () {
+    window.lio = {};
+  });
+
+  describe('of type "ordered"', function () {
+    it('should only show the first valid widget', function (done) {
+      var messageBar = new pathfora.Message({
+        id: 'messageBar1',
+        layout: 'bar',
+        msg: 'Welcome to our website'
+      });
+      var modal = new pathfora.Message({
+        id: 'modal2',
+        layout: 'modal'
+      });
+
+      pathfora.initializeWidgets([messageBar, modal], null, {
+        priority: 'ordered'
+      });
+
+      var widget1 = $('#' + messageBar.id);
+      var widget2 = $('#' + modal.id);
+
+      setTimeout(function () {
+        expect(widget1).toBeDefined();
+        expect(widget1.hasClass('opened')).toBeTruthy();
+        expect(widget2.length).toBe(0);
+        done();
+      }, 200);
+    });
+
+    it('should account for display conditions when determining priority', function (done) {
```

Comment:
probably not a terrible idea to test the inverse and make sure a modal shows based on display conditions also? probably overkill but might be worth testing both ends.

Reply:
This test is ensuring that it skips a widget with invalid display conditions and shows the first valid one. Not sure what you mean by testing the inverse? Are you saying I should put a valid display condition on the second widget? I'm not sure that would really test the priority code, but more of a display conditions test...

Reply:
yah i was more saying we should make sure that if it has valid display conditions we should be sure to not skip it but maybe that test is elsewhere?

Reply:
added